test/common/network/listener_impl_test.cc

```diff
@@ -154,5 +154,48 @@ TEST_P(ListenerImplTest, WildcardListenerUseActualDst) {
   dispatcher.run(Event::Dispatcher::RunType::Block);
 }
 
+TEST_P(ListenerImplTest, WildcardListenerIpv4Compat) {
+  Stats::IsolatedStoreImpl stats_store;
+  Event::DispatcherImpl dispatcher;
+  Network::TcpListenSocket socket(Network::Test::getAnyAddress(version_, true), true);
```

Comment:
I'd like a comment here (or at the top of the test) briefly explaining the circumstances this is testing.

It's confusing to me since this is the only place that leads to invoking Network::Utility::getAnyIpv6Address(false), so is this code testing a case that cannot occur outside of tests?

Reply:
This test mocks a case where a listener would be configured with an IPv6 any address and have it's `socket_address` configured with `ipv4_compat = true`. There are a couple of tests that do that, but in those the socket address comes from the listener configuraion, so they do not invoke Network::Utility::getAnyIpv6Address() at all. The effect is the same, though, when the listener is configured with the IPv6 ANY address ([::]).